doc/rfcs/004-quickly-collaborate-with-coworkers-and-friends.md

```diff
@@ -0,0 +1,166 @@
+
+## Motivation
+
+We hope to encourage more collaboration by reducing the barriers to entry. Specifically, we want to reduce the number of steps that exist between A) deciding that you *want* to collaborate and B) *actually* collaborating.
+
+Today, the transition from A to B involves the following steps:
+
+1. Host shares a portal
+2. Host copies portal URL to their clipboard
+3. Host switches from Atom to third-party communication tool (e.g., Slack, IRC)
+4. Host pastes portal URL into third-party communication tool
+5. Guest clicks portal URL, and the operating system hands control to Atom, and Atom joins the portal
+
+With the ability to invite trusted collaborators to your portal from within Atom, we reduce the set-up process from 5 steps to just 3 steps:
+
+1. Host selects a person in their list of trusted collaborators
+2. Host invites the selected person to a portal, and Atom automatically shares a portal for the host and invites the selected person
+3. Guest accepts invitation to join portal, and Atom joins the portal
+
+## Explanation
+
+### Accept portal invitations from a trusted collaborator
+
+Teletype provides a list of your recent collaborators. Each time a guest joins your portal, Teletype adds them to your list of recent collaborators. Each time you join a portal, Teletype adds the host to your list of recent collaborators.
+
+You can select a past collaborator and inform Teletype that you're willing to allow them to see when you're online [A], and that you're willing to receive portal invitations from them directly within Atom, thus identifying them as a "trusted collaborator."
+
+![](https://user-images.githubusercontent.com/378023/38798126-a2ccb7f8-419b-11e8-92d2-df0b57f311ec.png)
+
+## Invite a trusted collaborator to your portal
+
+Teletype presents your list of trusted collaborators sorted alphabetically by username.
+
+If you have designated a person as a trusted collaborator, and they have also designated you as a trusted collaborator, then you can see whether they're online [A], and they can see whether you're online.
+
+You can select a trusted collaborator that is currently online (an "invitee" in this context) and invite them to join your portal. If you're not already hosting a portal, Teletype automatically creates a portal for your workspace.
+
+![](https://user-images.githubusercontent.com/378023/38798163-c1dc4226-419b-11e8-9b96-5db9a9c90221.png)
+
+If the invitee chooses to join the portal, the portal opens in the invitee's workspace (just as it does when joining a portal via a URL).
+
+If the invitee declines to join the portal, Teletype notifies you that the guest declined your invitation. You cannot re-invite that invitee to the same portal. (There's probably a reason why they didn't join, so there's no need to bug them even more. If you think they declined by accident, reach out to them via a third-party chat service and send them the URL to join your portal.)
```

Comment:
~~I would prefer that Teletype does not inform you that a guest has declined your invitation, as part of the "negative feedback loop" I mentioned in 📹 . Harassers often view these "negative" events as a badge of honor (i.e. some people celebrate being blocked on Twitter as a sign they got under their target's skin).~~

~~If you are already relying on a third party chat application, I don't think it's necessary to send a "decline" back to the host.~~

EDIT: I didn't notice that this is in the "trusted collaborator" section. I think it would be important to distinguish that we ONLY do this for trusted collaborators, or have the ability to have pre-filled reasons to decline an invitation (i.e. "Sorry, I'm busy right now") or an away message (which is outlined as out of scope below). I think it would be important to clarify we do not send back a decline message if the two users are not mutually trusted collaborators.

Reply:
> I think it would be important to clarify we do not send back a decline message if the two users are not mutually trusted collaborators.

If the two users are not mutually-trusted collaborators, they can't send invitations to each other at all, which means that there's no invitation to decline. 😇

Reply:
Gotcha! Thanks for the clarification